components/button/button-add.js

```diff
@@ -59,30 +79,131 @@ class ButtonAdd extends PropertyRequiredMixin(FocusMixin(LocalizeCoreElement(Lit
 			button:${unsafeCSS(getFocusPseudoClass())} .line {
 				border-top-color: var(--d2l-color-celestine);
 			}
+		`;
+	}
 
-			.content {
+	constructor() {
+		super();
+
+		this.hintNearbyDistance = DEFAULT_HINT_DISTANCE;
+		this.hintTiming = HINT_TIMING.ALWAYS;
+		this.textVisible = false;
+
+		this._buttonId = getUniqueId();
+		this._hasFocusOrHover = false;
+		this._visibleDistance = 0;
+
+		this.addEventListener('focus', this._onFocusHover);
+		this.addEventListener('mouseover', this._onFocusHover);
+		this.addEventListener('blur', this._onBlur);
+		this.addEventListener('mouseout', this._onBlur);
+	}
+
+	static get focusElementSelector() {
+		return 'button';
+	}
+
+	render() {
+		const buttonSpacing = this._getButtonSpacing();
+		const text = this.text || this.localize('components.button-add.addItem');
+		const content = this.textVisible ? this._renderWithTextVisible(text) : this._renderWithTextHidden(text);
+		const id = !this.textVisible ? this._buttonId : undefined;
+
+		return html`
+			<button
+				class="d2l-label-text d2l-visible-on-ancestor-target"
+				id="${ifDefined(id)}"
+				style="${styleMap(buttonSpacing)}"
+				type="button">
+				<div class="line"></div>
+				${content}
+			</button>
+		`;
+	}
+
+	_getButtonSpacing() {
+		if (this.textVisible || this.hintTiming !== HINT_TIMING.NEARBY) return {};
+
+		this._visibleDistance = this.hintNearbyDistance || DEFAULT_HINT_DISTANCE;
```

Comment:
`this._visibleDistance` is used for calculating the tooltip offset

Reply:
Does this need to be a variable stored on the object? A couple other ways that might be nicer:

* custom `get/set` for `hintNearbyDistance` reactive property (naming to be determined), or
* calculating the value in `render` and then passing into `_getButtonSpacing` and `_renderWithTextHidden`, or
* moving this to small helper function `_getHintDistance()`, or 
* if it _does_ need to be separate variable stored on the object, then calculating this in [willUpdate](https://lit.dev/docs/v2/components/lifecycle/#willupdate) since it would only really need to be recalculated when  the `hintNearbyDistance` reactive property changes, or

Thoughts?